### PR TITLE
Add unit tests for bobbit.history

### PR DESCRIPTION
**Pull request by shawalli**
This PR adds unit tests for `bobbit.history`. Addresses task in #17.

### CODE REVIEW

File: src/bobbit/history.py [hunk 1]
```diff
@@ -22,9 +22,9 @@ def search(self, channel, nick=None, pattern=None, limit=None, reverse=False):
         if pattern:
             pattern = re.compile(pattern)
 
-        messages = self.history[channel]
+        messages = self.history.get(channel, collections.deque())
         if reverse:
-            messages = reversed(messages)
+            messages.reverse()
```
Comment (pbui) [inline src/bobbit/history.py]: I don't think we want to perform the reverse in place as that will permanently change the order of the deque in the history.  I think we want to get a new list.

**Comment by pbui** [comment]
@shawalli Besides the in-place reverse, everything else looks OK.  Go ahead and revert that and I'll test it out.  Thanks.

**Comment by shawalli** [comment]
The original implementation was throwing an error when I ran it, that's why
I changed it. Would you want me to copy the deque and reverse that? Or
perform the for loop in reverse order?

**Comment by pbui** [comment]
I think we should do:

```
 messages = list(reversed(messages))
```

File: tests/test_history.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        self.assertEqual(matches[0].body, "arbax")
+        self.assertEqual(matches[1].body, "baz")
+
+    async def test_match_limit(self):
+        matches = []
+        for m in self.hs.search(self.TEST_CHANNEL1, nick=self.TEST_USER2, limit=1):
+            matches.append(m)
+        
+        self.assertEqual(len(matches), 1)
+        self.assertEqual(matches[0].body, "baz")
```
Comment (pbui) [inline tests/test_history.py:135]: What do you think of adding the following:

```
if __name__ == '__main__':
    unittest.main()
```

That way we can run the test script without doing `python3 -m unittest ...`

Comment (shawalli) [reply tests/test_history.py:135]: Do you want that inside of that test or the `__init__.py` or somewhere else?

Comment (pbui) [reply tests/test_history.py:135]: At the bottom of `test_history.py`.

Comment (shawalli) [reply tests/test_history.py:135]: Sounds good, thanks for the clarification!

**Comment by pbui** [comment]
@shawalli Thank you for your contribution.

**Pull request merged**